src/components/Form/ToggleInput.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {
+  Component,
+  ReactElement,
+  ComponentClass,
+  FunctionComponent,
+} from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan } from '../../reset';
+import { ToggleProps, baseClassName, ToggleState } from './Toggle';
+
+const toggleDisabledClassName = `${baseClassName}--disabled`;
+
+const componentOrElementWithProps = (
+  component: ReactElement<any> | FunctionComponent<any> | ComponentClass<any>,
+  props: object,
+) => {
+  if (!!component) {
+    if (React.isValidElement(component)) {
+      return React.cloneElement(component, props); // element
+    }
+    return React.createElement(
+      component as FunctionComponent<any> | ComponentClass<any>,
+      props,
+    ); // component
+  }
+  return;
+};
+
+export class ToggleInput extends Component<ToggleProps> {
+  state: ToggleState = {
+    toggleState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  componentWillReceiveProps(nextProps: ToggleProps) {
+    const { checked } = nextProps;
```

Comment:
Is this implementation missing from the button-version?

Also could better rephrase `checked`-prop interface documentation:
`Controlled toggle-state - user actions use onClick to change`

Reply:
I added the missing implementation for button-version. And also rephrased the interface documentation.